metals/src/main/scala/scala/meta/internal/tvp/MetalsTreeViewProvider.scala

```diff
@@ -381,18 +404,34 @@ class FolderTreeViewProvider(
   def revealResult(
       path: AbsolutePath,
       closestSymbol: SymbolOccurrence,
-  ): Option[List[String]] =
+  ): Option[List[String]] = {
     if (path.isDependencySource(folder.path) || path.isJarFileSystem) {
-      buildTargets
+      def pathJar = AbsolutePath(
+        Paths.get(path.toNIO.getFileSystem().toString())
+      ).dealias
+      def jdkSources = JdkSources(userConfig().javaHome).toOption
+        .collect {
+          case sources
+              if sources == pathJar || !path.isJarFileSystem &&
+                path.isSrcZipInReadonlyDirectory(folder.path) =>
+            libraries.toUri(sources, closestSymbol.symbol).parentChain
+        }
+
+      val result = buildTargets
         .inferBuildTarget(List(Symbol(closestSymbol.symbol).toplevel))
         .map { inferred =>
-          libraries.toUri(inferred.jar, inferred.symbol).parentChain
+          val sourceJar = inferred.jar.parent.resolve(
+            inferred.jar.filename.replace(".jar", "-sources.jar")
+          )
+          libraries.toUri(sourceJar, inferred.symbol).parentChain
```

Comment:
You reveal here for toplevel and in `jdkSources` and workspace for found closest symbol. Maybe for `jdkSources` we should use the toplevel too?

Reply:
Changed, but most likely it would work ok anyway.